muxink/src/codec/bincode.rs

```diff
@@ -78,7 +82,9 @@ where
     type Output = T;
 
     fn transcode(&mut self, input: R) -> Result<Self::Output, Self::Error> {
-        bincode::deserialize(input.as_ref())
+        DefaultOptions::new()
```

Comment:
I'd factor out the `DefaultOptions` into 

```rust
fn mk_bincode() -> impl bincode::config::Options {
    DefaultOptions::new().reject_trailing_bytes()
}
```

as this prevents encoding and decoding options from accidentally drifting apart.

Reply:
I made the function name more explicit, because we actually need a different options for the `decode_frame`.

Surprisingly, the `allow_trailing_bytes()` in `decode_frame` seems redundant, it works correctly even with the `reject_trailing_bytes()` set (in case of which I'd expect the `codec::bincode::tests::decodes_frame` to fail since there **are** more bytes in the buffer.). I left the `allow_trailing_bytes()` intact for clarity of the intent (similarly to the redundant `with_varint_encoding()`).

Updated in 49a507c1d4cdcfc759fdffda2ab6f66d8147067d

Reply:
Superb, thanks!